core/eval.go

```diff
@@ -189,12 +193,37 @@ func evalSET(args []string, store *Store) []byte {
 				return RespNIL
 			}
 		case constants.NX:
+			nxOption = true
+
 			obj := store.Get(key)
 			if obj != nil {
 				return RespNIL
 			}
 		case constants.KEEPTTL, constants.Keepttl:
 			keepttl = true
+		case "GET":
+			 obj := store.Get(key)
+
+			 store.Put(key, store.NewObj(value, exDurationMs, oType, oEnc))
+
+			 if xxOption {
+				//store.Put(key, store.NewObj(value, exDurationMs, oType, oEnc))
+				if obj == nil {
+					return RespNIL
+				}
+				
+				return Encode(obj.Value, false)
+			 } else if nxOption {
+				//store.Put(key, store.NewObj(value, exDurationMs, oType, oEnc))
+				if obj == nil {
+					return Encode(obj.Value, false)
```

Comment:
Shouldn't this return RespNIL?
If key `k` doesn't exit and you fire `SET k v NX GET` the result will be `(nil)`.

```
redis> SET k v NX GET
(nil)
```

Also, add a test for this once fixed.